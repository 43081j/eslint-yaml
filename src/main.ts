import type {ESLint} from 'eslint';
import {YAMLanguage} from './languages/yaml-language.js';
import {YAMLSourceCode} from './languages/yaml-source-code.js';
import {createConfig as createRecommendedConfig} from './configs/recommended.js';
import {rules} from './rules/all.js';

const plugin: ESLint.Plugin = {
  meta: {
    name: 'eslint-yaml',
    version: '0.0.1'
  },
  languages: {
    yaml: new YAMLLanguage()
  },
  rules
};

plugin.configs = {
  recommended: createRecommendedConfig(plugin)
};

export default plugin;
export {YAMLSourceCode};
export * from './languages/yaml-language.js';
