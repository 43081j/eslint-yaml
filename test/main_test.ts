import {ESLint} from 'eslint';
import {expect, describe, it, beforeEach} from 'vitest';
import * as yaml from '../src/main.js';

describe('main', () => {
  it('should export correct shape', () => {
    expect(yaml.yaml).toBeDefined();
    expect(yaml.YAMLSourceCode).toBeDefined();
    expect(yaml.YAMLLanguage).toBeDefined();
    expect(Object.keys(yaml.yaml)).toEqual([
      'meta',
      'languages',
      'rules',
      'configs'
    ]);
  });

  describe('eslint integration', () => {
    let eslint: ESLint;

    beforeEach(() => {
      eslint = new ESLint({
        overrideConfigFile: true,
        overrideConfig: {
          files: ['**/*.yaml'],
          plugins: {yaml: yaml.yaml},
          language: 'yaml/yaml',
          rules: {
            'yaml/no-empty-mapping-value': 'error'
          }
        }
      });
    });

    it('should raise diagnostics', async () => {
      const result = await eslint.lintText('key:', {filePath: 'test.yaml'});
      expect(result[0].messages).toMatchSnapshot();
    });
  });
});
