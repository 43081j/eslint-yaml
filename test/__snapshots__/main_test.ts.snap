// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`main > eslint integration > should raise diagnostics 1`] = `
[
  {
    "column": 1,
    "endColumn": 5,
    "endLine": 1,
    "line": 1,
    "message": "Empty mapping value is not allowed.",
    "messageId": "emptyMappingValue",
    "nodeType": "Pair",
    "ruleId": "yaml/no-empty-mapping-value",
    "severity": 2,
  },
]
`;

exports[`main > should correctly set recommended config 1`] = `
{
  "yaml/block-mapping-colon-indicator-newline": "error",
  "yaml/no-empty-mapping-value": "error",
}
`;
